pages/admin.js

```diff
@@ -0,0 +1,143 @@
+import Head from "next/head";
+import db from "../data/database";
+import { products as Products } from "../data/products";
+import { ListProductComponent } from "../components/product/list";
+import { useState, useEffect, useCallback } from "react";
+
+
+export default function Admin() {
+  console.log(Products)
+  let [product, setProduct] = useState({
```

Comment:
if you're not familiar with the `useReducer` hook, please look that up. that hook is more suitable for this kind of advanced data structure of states.